### PR TITLE
Fix Delta Lake queries with predicates not expressible by tuple domain

### PR DESCRIPTION
**Pull request by homar**
It didn't work when table was partitioned by column that contained null values
and predicate was applied to it
## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java [hunk 1]
```diff
@@ -184,9 +184,9 @@ private Stream<DeltaLakeSplit> getSplits(
                                 .filter(column -> partitionValues.containsKey(((DeltaLakeColumnHandle) column).getName()))
                                 .collect(toImmutableMap(identity(), column -> {
                                     DeltaLakeColumnHandle deltaLakeColumn = (DeltaLakeColumnHandle) column;
-                                    return NullableValue.of(
+                                    return new NullableValue(
                                             deltaLakeColumn.getType(),
-                                            deserializePartitionValue(deltaLakeColumn, addAction.getCanonicalPartitionValues().get(deltaLakeColumn.getName())));
+                                            deserializePartitionValue(deltaLakeColumn, partitionValues.get(deltaLakeColumn.getName())));
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java:189]: nit: for a minimal change you could keep `addAction.getCanonicalPartitionValues()` here 
(or make a separate commit like in https://github.com/trinodb/trino/pull/12052: https://github.com/trinodb/trino/pull/12052/commits/a03b205450530cb9741e5176cc8b1eec65d8cbe4)

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDelete.java [hunk 2]
```diff
@@ -270,6 +270,19 @@ public void testDeleteMultiplePartitionKeys()
         assertQuery("SELECT * FROM " + tableName, "VALUES (1, 2, 4), (3, 2, 1), (null, null, null), (1, 1, 1)");
     }
 
+    // This is not the best class for this test but it doesn't fit any other places
+    // and here entire test infrastructure is present and ready
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDelete.java:274]: This is a discourse between code author and a reviewer. Doesn't need to be persisted in the source code comment.

However, indeed, the class is about deletes and we don't do any deletes here.
let's move it elsewhere, perhaps into BaseDeltaLakeMinioConnectorTest

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDelete.java [hunk 3]
```diff
@@ -270,6 +270,19 @@ public void testDeleteMultiplePartitionKeys()
...
+    @Test
+    public void testQueryingNullPartitions()
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDelete.java:276]: `testQueryNullPartitionWithNonPushdownablePredicate` + add a comment that non-pushdownable means not covertiabe to a TupleDomain

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java (hunk 1 above)
Comment (homar) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java:189]: ok I will remove it. I changed it because it hurt my eyes..

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDelete.java (hunk 2 above)
Comment (homar) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDelete.java:274]: I wanted to avoid the discussion and leave it for future maintainers ;)

**Review by findepi — APPROVED** [review]
